tests/_core/test_spatialdata_operations.py

```diff
@@ -115,35 +115,28 @@ def test_concatenate_tables():
     d["region"] = sorted(d["region"])
     assert d == {
         "region": ["shapes/circles", "shapes/poly"],
-        "region_key": "annotated_element_merged_1",
+        "region_key": "region",
         "instance_key": "instance_id",
     }
 
-    ##
     table3 = _get_table(region="shapes/circles", region_key="annotated_shapes_other", instance_key="instance_id")
-    table3.uns[TableModel.ATTRS_KEY]["region_key"] = "annotated_shapes_other"
-    with pytest.raises(AssertionError):
-        _concatenate_tables([table0, table3])
-    table3.uns[TableModel.ATTRS_KEY]["region_key"] = None
-    table3.uns[TableModel.ATTRS_KEY]["instance_key"] = ["shapes/circles", "shapes/poly"]
-    with pytest.raises(AssertionError):
-        _concatenate_tables([table0, table3])
-
-    ##
+    with pytest.raises(ValueError, match="Categorical categories must be unique"):
+        _concatenate_tables([table0, table3], region_key="region")
+
     table4 = _get_table(
-        region=["shapes/circles", "shapes/poly"], region_key="annotated_shape0", instance_key="instance_id"
+        region=["shapes/circles1", "shapes/poly1"], region_key="annotated_shape0", instance_key="instance_id"
     )
     table5 = _get_table(
-        region=["shapes/circles", "shapes/poly"], region_key="annotated_shape0", instance_key="instance_id"
+        region=["shapes/circles2", "shapes/poly2"], region_key="annotated_shape0", instance_key="instance_id"
     )
     table6 = _get_table(
-        region=["shapes/circles", "shapes/poly"], region_key="annotated_shape1", instance_key="instance_id"
+        region=["shapes/circles3", "shapes/poly3"], region_key="annotated_shape1", instance_key="instance_id"
+    )
+    with pytest.raises(ValueError, match="`region_key` must be specified if tables have different region keys"):
+        assert len(_concatenate_tables([table4, table5, table6])) == len(table4) + len(table5) + len(table6)
```

Comment:
Why is there an assertion in here?

The assertion is never executed, so I'm not seeing the point in including it.

Reply:
my mistake